support-frontend/assets/pages/digital-subscription-landing/components/paymentSelection/helpers/paymentSelection.jsx

```diff
@@ -93,12 +110,12 @@ const BILLING_PERIOD = {
 
 // state
 const mapStateToProps = (state: State): { paymentOptions: Array<PaymentOption> } => {
-  const { productPrices } = state.page;
+  const { productPrices, orderIsAGift } = state.page;
   const { countryGroupId, currencyId } = state.common.internationalisation;
   const productOptions = getProductOptions(productPrices, countryGroupId);
 
   const createPaymentOption = (billingPeriod: BillingPeriod): PaymentOption => {
-    const digitalBillingPeriod = billingPeriod === 'Monthly' || billingPeriod === 'Annual' ? billingPeriod : 'Monthly';
+    const digitalBillingPeriod = billingPeriod === 'Monthly' || billingPeriod === 'Annual' || billingPeriod === 'Quarterly' ? billingPeriod : 'Monthly';
```

Comment:
Thanks, much nicer!

Reply:
However, this had made Flow very sad. I think it makes Flow realise that we're leaving out 'SixWeekly' which is part of the type 'BillingPeriod' but not part of the digital subscriptions setup. There's something quite strange going on with the types in this component, partly because of code that it shares with other subscriptions products that handle all types of products and thus their various billing periods. Given this, I am planning to leave this piece of code as it is for the moment. I will put in a pairing candidate card to refactor the payment selection component.